app/android/runtime_client/BUILD.gn

```diff
@@ -0,0 +1,57 @@
+# Copyright (c) 2016 Intel Corporation. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+import("//build/config/android/rules.gni")
+import("//xwalk/build/version.gni")
+
+android_library("xwalk_app_runtime_java") {
+  java_files = [
+    "//xwalk/app/android/runtime_activity/src/org/xwalk/app/XWalkRuntimeActivityBase.java",
+    "src/org/xwalk/app/runtime/extension/XWalkRuntimeExtensionLoader.java",
+    "src/org/xwalk/app/runtime/XWalkCoreProviderImpl.java",
+    "src/org/xwalk/app/runtime/XWalkRuntimeTestHelper.java",
+    "src/org/xwalk/app/runtime/XWalkRuntimeView.java",
+    "src/org/xwalk/app/runtime/XWalkRuntimeViewProvider.java",
+    "src/org/xwalk/app/runtime/XWalkRuntimeViewProviderFactory.java",
+  ]
+  deps = [
+    "//xwalk/runtime/android/core:xwalk_core_java",
+  ]
+  srcjar_deps = [ ":xwalk_runtime_client_version" ]
+}
+
+copy("xwalk_runtime_client_version") {
```

Comment:
These two actions are also too complicated for no reason and another instance of "look at the bigger picture and don't just copy gyp".

First of all, `generate_runtime_client_version.py` is an ugly, useless script that has no reason to exist. I'm removing it in #3878.

Secondly, GN has a much better infrastructure to help you not reinvent the wheel: `chrome/version.gni` and `build/config/zip.gni`. Look at `components/cronet/android/BUILD.gn` to see what I'm talking about. The whole file could be rewritten like this:

```
import("//build/config/zip.gni")
import("//chrome/version.gni")

android_library("xwalk_app_runtime_java") {
# ...
}

_version_dir =
    "$target_gen_dir/templates/xwalk_runtime_client_version_java"
_version_file =
    "$_version_dir/org/xwalk/app/runtime/XWalkRuntimeClientVersion.java"

process_version("runtime_version_java") {
  visibility = [ ":*" ]
  process_only = true
  # sources = [ "//xwalk/VERSION" ]
  extra_args = [ "-f", rebase_path("//xwalk/VERSION") ]
  template_file = "src/templates/XWalkRuntimeClientVersion.template"
  output = _version_file
}

zip("runtime_version_srcjar") {
  visibility = [ ":*" ]
  inputs = [ _version_file ]
  output = "$target_gen_dir/$target_name.srcjar"
  base_dir = _version_dir
  deps = [ ":runtime_version_java" ]
}

```

Note that there's a commented out line in `runtime_version_java`. That's because `version.gni` passes some files such as `chrome/VERSION` to version.py after the files listed in `sources`, so `chrome/VERSION` will override the values in `xwalk/VERSION`. I'm trying to change the order in https://codereview.chromium.org/2308583002, but until then you can use the version that is not commented out that achieves the same final result.